keps/sig-scheduling/785-scheduler-component-config-api/README.md

```diff
@@ -0,0 +1,110 @@
+# KEP-785: Scheduler Component Config API
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [x] Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [x] KEP approvers have approved the KEP status as `implementable`
+- [x] Design details are appropriately documented
+- [x] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [x] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The kube-scheduler configuration API `kubescheduler.config.k8s.io` is currently
+in version `v1alpha2`. We propose its graduation to `v1beta1` in order to
+promote its wider use.
+
+## Motivation
+
+The `kubescheduler.config.k8s.io` API has been in alpha stage for several
+releases. In release 1.18, we introduced `v1alpha2`, including important
+changes such as:
+
+- The removal of the old Policy API in favor of plugin configurations, that
+  align with the new scheduler framework.
+- The introduction of scheduling profiles, that allow a scheduler to appear
+  as multiple schedulers under different configurations.
+  
+A configuration API allows cluster administrators to build, validate and
+version their configurations in a more robust way than using command line flags.
+
+Graduating this API to Beta is a sign of its maturity that would encourage wider
+usage.
+
+### Goals
+
+- Introduce `kubescheduler.config.k8s.io/v1beta1` as a copy of
+`kubescheduler.config.k8s.io/v1alpha2` with minimal cleanup changes.
+- Use the newly created API objects to build the default configuration for kube-scheduler.
+
+### Non-Goals
+
+- Remove support for `kubescheduler.config.k8s.io/v1alpha2`
```

Comment:
we still can remove v1alpha2, right?

Reply:
Yes, but it is outside of the scope of this KEP and thus, this release.

Reply:
why don't we remove it in this release?

Reply:
I would remove it in this release, k8s depreciation policy allows that, it is even listed in the examples. We kept v1alpha1 in the previous release mainly because the changes were significant, this is a lot less so.

Reply:
Thoughts, @liggitt?

Reply:
We can get more opinions, but the policy allows it and it is up to the sig to decide what to do here.

Reply:
> the policy allows it and it is up to the sig to decide what to do here

Correct.

Reply:
ok, I'll craft a patch, as assumptions would change.